slsqp/slsqp.c

```diff
@@ -2044,12 +2044,16 @@ static void slsqpb_(int *m, int *meq, int *la, int *
     *mode = line == 1 ? -2 : 1;
     goto L330;
 L200:
-    if (h1 <= h3 / ten || line > 10) {
-	goto L240;
+    if (isfinite(h1)) {
```

Comment:
`isfinite` is a C99 function, and right now NLopt is not requiring C99.   So, probably we need a check (similar to the check for `isinf`) in `configure.ac` that defines `HAVE_ISFINITE` if `isfinite` is available,  a `nlopt_isfinite(x)` function in `api/general.c` that calls `isfinite` if it is available and otherwise uses its own implementation, and a declaration of `nlopt_isfinite` in `util/nlopt-util.h`

e.g. doing `#include <float.h>` and doing `nlopt_isfinite(double x) { return fabs(x) <= DBL_MAX; }` should work in any C89 compiler.